source/includes/elastic-agents/_031-list-agents.md

```diff
@@ -1,55 +1,56 @@
 ## List Agents
 
 ```java
-import com.thoughtworks.go.plugin.api.*;
-import com.thoughtworks.go.plugin.api.annotation.Extension;
-import com.thoughtworks.go.plugin.api.logging.Logger;
-import com.thoughtworks.go.plugin.api.request.*;
-import com.thoughtworks.go.plugin.api.response.*;
-import com.google.gson.Gson;
-import java.util.*;
-
-@Extension
-public class DockerPlugin implements GoPlugin {
-  private GoApplicationAccessor accessor;
-  public static final Logger LOG = Logger.getLoggerFor(DockerPlugin.class);
-
-  public void initializeGoApplicationAccessor(GoApplicationAccessor accessor) {
-    this.accessor = accessor;
-  }
-
-  public GoPluginIdentifier pluginIdentifier() {
-    return new GoPluginIdentifier("elastic-agent", Arrays.asList("1.0"))
-  }
-
-  private List listAgents() {
-    // create a request
-    DefaultGoApiRequest request = new DefaultGoApiRequest(
-      "go.processor.elastic-agents.list-agents",
-      "1.0",
-      pluginIdentifier()
-    );
-
-    // submit the request
-    GoApiResponse response = accessor.submit(request);
-
-    // check status
-    if (response.responseCode() != 200) {
-      LOG.error("The server sent an unexpected status code " + response.responseCode() + " with the response body " + response.responseBody());
+
+import com.thoughtworks.go.plugin.api.GoApplicationAccessor;
+import com.thoughtworks.go.plugin.api.request.DefaultGoApiRequest;
+import com.thoughtworks.go.plugin.api.response.GoApiResponse;
+import java.util.Collection;
+import static cd.go.contrib.elasticagents.docker.Constants.API_VERSION;
+import static cd.go.contrib.elasticagents.docker.Constants.PLUGIN_IDENTIFIER;
+
+public class PluginRequest {
+    private final GoApplicationAccessor accessor;
+
+    public PluginRequest(GoApplicationAccessor accessor) {
+        this.accessor = accessor;
+    }
+
+    public PluginSettings getPluginSettings() throws ServerRequestFailedException {
+        DefaultGoApiRequest request = new DefaultGoApiRequest(Constants.REQUEST_SERVER_GET_PLUGIN_SETTINGS, API_VERSION, PLUGIN_IDENTIFIER);
```

Comment:
You seem to have copy pasted code, and there's no understanding of what `Constants.REQUEST_SERVER_GET_PLUGIN_SETTINGS` or `API_VERSION` or `PLUGIN_IDENTIFIER` is. It was intentionally inlined, to serve as a example that is simple to follow and understand.